### PR TITLE
Add traverse support for dataclasses

### PR DESCRIPTION
**Pull request by aberres**
`Dataclasses` are not traversed right now. As I see them replacing pure `dicts` in quite some cases I think it would be nice to handle them as well.

No tests yet added, wanted to check first if you guys are interested.

- [x] Tests added / passed
- [x] Passes `flake8 dask`

### CODE REVIEW

**Comment by aberres** [comment]
Hm, seems as if this is incomplete.

While threaded execution works using processes fails when it tries to pickle `Delayed` instances.
Seems as if I am something.

**Comment by aberres** [comment]
Needed to scratch that itch... 

Seems to work with the latest update.

**Comment by mrocklin** [comment]
@jcrist can I ask you to review and manage this proposal?

File: dask/base.py [hunk 1]
```diff
@@ -10,6 +10,11 @@
 import threading
 import uuid
 
+try:
+    import dataclasses
+except ImportError:
+    dataclasses = None
```
[the remark this answers is not in the document]
Comment (aberres) [reply dask/base.py]: Done.

**Comment by aberres** [comment]
The test in `test_base.py` has a `try: ... except ImportError: ` as otherwise the test fails due to the `dataclass` not coming from a module level scope.

**Pull request merged**